File: backend/src/main/java/com/campus/marketplace/entity/Message.java
package com.campus.marketplace.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("messages")
public class Message {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long conversationId;

    private Long senderId;

    private String content;

    private Integer type;

    private BigDecimal priceOffer;

    private Integer offerStatus;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
}

File: frontend/types/message.ts
export interface Conversation {
  id: number
  user1Id: number
  user2Id: number
  productId: number
  lastMessage: string | null
  lastMessageAt: string | null
  createdAt: string
}

export interface Message {
  id: number
  conversationId: number
  senderId: number
  content: string
  type: number
  priceOffer: number | null
  offerStatus: number | null
  createdAt: string
}

export interface MessageSendRequest {
  conversationId: number
  content: string
  type?: number
  priceOffer?: number
}
